src/components/coupons/CouponCaption.tsx

```diff
@@ -0,0 +1,91 @@
+import { gql } from '@apollo/client'
+import { TypographyProps } from '@mui/material'
+
+import { intlFormatNumber } from '~/core/intlFormatNumber'
+import { CouponItemFragment, CouponTypeEnum, CouponFrequency } from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { Typography } from '~/components/designSystem'
+
+gql`
+  fragment CouponCaption on Coupon {
+    id
+    amountCurrency
+    amountCents
+    couponType
+    percentageRate
+    frequency
+    frequencyDuration
+  }
+`
+
+interface CouponCaptionProps {
+  coupon: CouponItemFragment
+  variant?: TypographyProps['variant']
+}
+
+export const CouponCaption = ({ coupon, variant = 'caption' }: CouponCaptionProps) => {
+  const { translate } = useInternationalization()
+
+  const getCaption = () => {
```

Comment:
You can `memo` this whole component to avoid re-rendering (`memo`, no `useMemo` :) )